src/neuroconv/tools/spikeinterface/spikeinterface.py

```diff
@@ -997,24 +997,35 @@ def write_recording(
         metadata = dict_deep_update(recording.nwb_metadata, metadata)
     elif metadata is None:
         metadata = get_nwb_metadata(recording=recording)
+
     with make_or_load_nwbfile(
         nwbfile_path=nwbfile_path, nwbfile=nwbfile, metadata=metadata, overwrite=overwrite, verbose=verbose
     ) as nwbfile_out:
-        add_all_to_nwbfile(
```

Comment:
This functions is working as a wrapper for 4 statements. 
As I see it, it is just adding docstring maintenance and lines costs whereas not really providing any useful abstraction and in fact being opaque on its purpose ( not clear what "all" is on "add_all_to_nwbfile"). 

I am writing the whole logic of `write_segmentation` bare at the function as I think is easier to understand and maintain in this way, let me know if I am missing something.

Relevant:
https://matthewrocklin.com/blog/work/2019/06/23/avoid-indirection

Reply:
Yeah that's fine